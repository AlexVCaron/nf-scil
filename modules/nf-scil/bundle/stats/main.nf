process BUNDLE_STATS {
    tag "$meta.id"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_2.0.2.sif':
        'scilus/scilus:2.0.2' }"

    input:
    tuple val(meta), path(bundles), path(metrics), path(endpoints_head), path(endpoint_tail), path(voxel_labels_map), path(labels_map), path(distance_map)

    output:
    tuple val(meta), path("*.json"), emit: output
    
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    
    """
    

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bundle: \$(samtools --version |& sed '1!d ; s/samtools //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    
    """
    

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bundle: \$(samtools --version |& sed '1!d ; s/samtools //')
    END_VERSIONS
    """
}
