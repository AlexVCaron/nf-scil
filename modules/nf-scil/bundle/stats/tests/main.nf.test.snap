{
    "bundle - stats": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__length_stats.json:md5,72873da12072e254f369ec316911df28"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__endpoints_map_raw.json:md5,2a86a4d10878c942c09723c3122940b4"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__endpoints_metric_stats.json:md5,d13e29a5d9dd09e8b5a2d40b3b61b3a9"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__mean_std.json:md5,f7463e5519063f7e0f32709d0dfdc1a2"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__volume.json:md5,f2cf5bbf678b0d364f26120ad10900ae"
                    ]
                ],
                "5": [
                    
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__volume_per_label.json:md5,1ba8e9d51cc2dec86fdaa4498488424b"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__mean_std_per_point.json:md5,efcac000fb6c0f607b7377154e68854c"
                    ]
                ],
                "8": [
                    
                ],
                "9": [
                    "versions.yml:md5,4eb4fdf7390c520819c0084762c6adc6"
                ],
                "endpoints_metric": [
                    
                ],
                "endpoints_metric_stats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__endpoints_metric_stats.json:md5,d13e29a5d9dd09e8b5a2d40b3b61b3a9"
                    ]
                ],
                "endpoints_raw": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__endpoints_map_raw.json:md5,2a86a4d10878c942c09723c3122940b4"
                    ]
                ],
                "length": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__length_stats.json:md5,72873da12072e254f369ec316911df28"
                    ]
                ],
                "mean_std": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__mean_std.json:md5,f7463e5519063f7e0f32709d0dfdc1a2"
                    ]
                ],
                "mean_std_per_point": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__mean_std_per_point.json:md5,efcac000fb6c0f607b7377154e68854c"
                    ]
                ],
                "streamline_count": [
                    
                ],
                "versions": [
                    "versions.yml:md5,4eb4fdf7390c520819c0084762c6adc6"
                ],
                "volume": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__volume.json:md5,f2cf5bbf678b0d364f26120ad10900ae"
                    ]
                ],
                "volume_per_labels": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__volume_per_label.json:md5,1ba8e9d51cc2dec86fdaa4498488424b"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-29T15:01:20.785499"
    },
    "bundle - stats - light": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__streamline_count.json:md5,432344fd244a5d72a63758af7e7df6bb"
                    ]
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "9": [
                    "versions.yml:md5,4eb4fdf7390c520819c0084762c6adc6"
                ],
                "endpoints_metric": [
                    
                ],
                "endpoints_metric_stats": [
                    
                ],
                "endpoints_raw": [
                    
                ],
                "length": [
                    
                ],
                "mean_std": [
                    
                ],
                "mean_std_per_point": [
                    
                ],
                "streamline_count": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__streamline_count.json:md5,432344fd244a5d72a63758af7e7df6bb"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4eb4fdf7390c520819c0084762c6adc6"
                ],
                "volume": [
                    
                ],
                "volume_per_labels": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-29T15:05:01.55552"
    }
}